### PR TITLE
migrate to environment variables over site config

### PR DESCRIPTION
**Pull request by coury-clark**
Site config is not trivial to manipulate in managed instances, and is not restricted to Sourcegraph employees (or license checks, for example). This migrates all the required config to environment variables.

## Test plan

```
EXPORT_USAGE_DATA_ENABLED=true EXPORT_USAGE_DATA_TOPIC_NAME=usage-data-testing EXPORT_USAGE_DATA_TOPIC_PROJECT=sourcegraph-dogfood gogo

[         worker] INFO worker.export-usage-telemetry telemetry/telemetry_job.go:196 fetching events from bookmark {"bookmark_id": 21816}
[         worker] INFO worker.export-usage-telemetry telemetry/telemetry_job.go:207 telemetryHandler executed {"event count": 4, "maxId": 21821}

```

### CODE REVIEW

File: cmd/frontend/envvar/envvar.go [hunk 1]
```diff
@@ -20,7 +20,6 @@ var HTTPAddrInternal = env.Get(
 
 var sourcegraphDotComMode, _ = strconv.ParseBool(env.Get("SOURCEGRAPHDOTCOM_MODE", "false", "run as Sourcegraph.com, with add'l marketing and redirects"))
 var openGraphPreviewServiceURL = env.Get("OPENGRAPH_PREVIEW_SERVICE_URL", "", "The URL of the OpenGraph preview image generating service")
-var exportUsageData, _ = strconv.ParseBool(env.Get("EXPORT_USAGE_DATA", "false", "Export usage data from this Sourcegraph instance to centralized Sourcegraph analytics (requires restart)."))
```
Comment (coury-clark) [inline cmd/frontend/envvar/envvar.go:23]: Having these in different places was proving confusing, so just migrating it all to the telemetry job package

File: enterprise/cmd/worker/internal/telemetry/telemetry_job_test.go [hunk 2]
```diff
@@ -95,21 +91,15 @@ func TestHandlerEnabledDisabled(t *testing.T) {
...
 			err := handler.Handle(ctx)
-			if err != nil {
-				if !errors.Is(err, disabledErr) {
-					t.Error("unexpected error from Handle function, expected disabled error")
-				}
-			} else {
-				if test.expectErr != nil {
-					t.Error("expected error but did not receive one")
-				}
+			if !errors.Is(err, test.expectErr) {
```
Comment (coury-clark) [inline enterprise/cmd/worker/internal/telemetry/telemetry_job_test.go:100]: This test actually had a bug that non-nil errors weren't checked against nil expectations

File: enterprise/cmd/worker/internal/telemetry/telemetry_job_test.go [hunk 3]
```diff
@@ -645,3 +623,19 @@ func initAllowedEvents(t *testing.T, db database.DB, names []string) {
 		t.Fatal(err)
 	}
 }
+
+func mockEnvVars(t *testing.T, flag bool) {
+	prevEnabled := enabled
```
Comment (coury-clark) [inline enterprise/cmd/worker/internal/telemetry/telemetry_job_test.go:628]: I tried using `t.setEnv` but unfortunately our environment variables package requires initialization at package level which throws off ordering here.

File: enterprise/cmd/worker/internal/telemetry/telemetry_job_test.go [hunk 4]
```diff
@@ -95,22 +91,25 @@ func TestHandlerEnabledDisabled(t *testing.T) {
...
-				}
-			} else {
-				if test.expectErr != nil {
-					t.Error("expected error but did not receive one")
-				}
+			if !errors.Is(err, test.expectErr) {
+				t.Errorf("unexpected error from Handle function, expected error: %v, received: %s", test.expectErr, err.Error())
+
 			}
+			// if err != nil {
```
Comment (chwarwick) [inline enterprise/cmd/worker/internal/telemetry/telemetry_job_test.go]: can we remove this or is it coming back?

**Review by chwarwick — APPROVED** [review]

File: enterprise/cmd/worker/internal/telemetry/telemetry_job_test.go (hunk 4 above)
Comment (coury-clark) [reply enterprise/cmd/worker/internal/telemetry/telemetry_job_test.go]: @chwarwick I think the latest commit already has that removed - can you verify?

**Pull request merged**